schemas/com.snowplowanalytics.monitor/application/jsonschema/1-0-0

```diff
@@ -0,0 +1,112 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for an application entity, attached as context to application status report events",
+  "self": {
+    "vendor": "com.snowplowanalytics.monitor",
+    "name": "application",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+
+  "type": "object",
+  "properties": {
+    "name": {
+      "description": "The name of the application, eg rdb-loader",
+      "type": ["string", "null"],
+      "maxLength": 256
+    },
+    "category": {
+      "description": "The category of the application, eg loader",
+      "type": ["string", "null"],
+      "maxLength": 256
+    },
+    "version": {
+      "description": "The version of the application, eg 1.22.1",
+      "type": ["string", "null"],
+      "maxLength": 32
+    },
+    "instance_id": {
+      "description": "The ID of the application instance; a UUID",
+      "type": "string",
+      "maxLength": 36
+    },
+    "instance_name": {
+      "description": "The name of the application instance, eg com-acme-prod1-rdb-loader-1",
+      "type": ["string", "null"],
+      "maxLength": 256
+    },
+    "inputs": {
```

Comment:
Hey @dilyand - when I discussed this with @jbeemster we realised that applications would only have one input. (I.e. an app will only consume a single stream.) However, it could have many outputs.

For that reason we thought have dedicated `input_id` and `input_type` string fields in teh schema, but have an `outputs` field that was an array of objects where each object contains an `id` field (UUID)  and `type` field (string)

Reply:
Looking further down the schema that's exactly how you've specified the `outputs` field

Reply:
I thought it was better to structure the inputs in the same way as the outputs.

- If we only have one input, then the array will just have one value. I don't see a downside.
- If in future there's a use case with more than one input, then we'll just add another value to the array and won't need to upgrade the schema.
- This also supports a use case with an arbitrary number of inputs, potentially changing at different stages of the app execution.
- It's more consistent.

But happy to reconsider if this structure will make it harder to process and visualise the data.

Reply:
You've won me over! Thanks